test/bandit/http1/plug_test.exs

```diff
@@ -134,9 +134,15 @@ defmodule HTTP1PlugTest do
       SimpleHTTP1Client.send(client, "GET", "/hello_world", ["host: banana"])
       assert SimpleHTTP1Client.connection_closed_for_reading?(client)
 
-      assert_receive {:log, %{level: :error, msg: {:string, msg}}}, 500
+      assert_receive {:log, %{level: :error, msg: {:string, msg}, meta: meta}}, 500
       refute msg =~ "(Plug.Conn.WrapperError)"
-      assert msg =~ "** (RuntimeError) boom"
+      assert msg =~ "** (MatchError)"
```

Comment:
What's the reason for this change? Is there an issue that comes up with MatchError that doesn't with a simple raise?

Reply:
I see. I wonder if we should be doing that for all errors (ie continue to have the `Plug.Conn.WrapperError` pattern match be 'dumb' and just pass the value through unchanged) and normalize in the other handle_error clauses. Does that make sense, or is it only needed in the narrow case of wrapped errors?

Reply:
Yes, I think you're right.
That's regardless of whether it's wrapped or not.

Reply:
Updated.

Reply:
Also now realized this will fix/normalize the emitted telemetry exception.

Updated PR title.